### PR TITLE
Add TypeScript module support

### PR DESCRIPTION
**Pull request by ntsd**
Add TypeScript support

Please rebuild (npm run build) and dist (npm run dist) again.

Example use

```ts
import temml, { type Options } from 'temml';

const tex = "c = \\pm\\sqrt{a^2 + b^2}";
const temmlOptions: Options = {};
temml.renderToString(tex, temmlOptions);
```

more example: https://github.com/ntsd/svelte-tex/blob/main/src/lib/MathML.svelte

Node prefers to add more types.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,9 @@
   },
   "exports": {
     ".": {
-      "require": "./dist/temml.cjs"
+      "require": "./dist/temml.cjs",
+      "import": "./dist/temml.cjs",
```
Comment (Trildar) [inline package.json:12]: I think this should point to the `mjs` file as `import` typically implies ESM or at least some build system that can handle ESM.

File: dist/temml.d.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+export interface Options {
+  displayMode?: boolean;
+  annotate?: boolean;
+  leqno?: boolean;
+  throwOnError?: boolean;
+  errorColor?: string;
+  macros?: Record<string, string>;
+  wrap?: "tex" | "=";
```
Comment (ronkok) [inline dist/temml.d.ts:8]: Please revise to `wrap?: "tex" | "=" | "none";` 

`none` is a valid value. I'll update the documentation in the next release.

File: temml.d.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+export interface Options {
+  displayMode?: boolean;
+  annotate?: boolean;
+  leqno?: boolean;
+  throwOnError?: boolean;
+  errorColor?: string;
+  macros?: Record<string, string>;
+  wrap?: "tex" | "=";
```
Comment (ronkok) [inline temml.d.ts:8]: Again, please revise to `wrap?: "tex" | "=" | "none";`

**Review by ronkok — CHANGES_REQUESTED** [review]
@ntsd, Thank you for this PR. I don't know a lot about TypeScript and I am grateful for the help. Also, thank you for using the JSDoc comments. I find that to be much superior to inline type syntax.

I've made a couple of comments. When they are picked up, this should be ready to go.

**Comment by ntsd** [comment]
@ronkok, I did update the wrap type and defined the export functions type (still have `any` type but should have no problem).

Also, the commit includes JSDoc on the dist folder.

I am using Temmel on this module https://github.com/ntsd/svelte-tex the type is working fine.

**Comment by ronkok** [comment]
Looks good to me. Thank you, @ntsd  and @Trildar.

**Pull request merged**